### DISCUSSION TITLE
Support for Schema Invalidation

### DISCUSSION THREAD

**Discussion by jimitndiaye** [ChilliCream/hotchocolate#5447] [Q&A] [answered]
I have a rather unusual scenario in which I'm generating a graphql schema over a data source whose structure may change periodically (e.g fields being added/removed etc). 

Not being familiar with Hot Chocolote's inner workings (I've only used graphql-dotnet previously) I'm not sure how to go about invalidating the schema of a running graphql server instance so that it can be rebuilt on the next request, for instance using a cache.
In graphql-dotnet the schema instance is a standalone that can be registered with DI or cached and thus invalidated. In HC you instead build an IRequestExecutor and there isn't an obvious way to get at the schema itself.

I've thought of an alternative way that instead builds the invalidation into the resolvers but I'm not sure if the request executor would realize that a graphql type has changed and update any introspection responses accordingly.

**Comment by tobias-tengler** [comment, marked as answer]
Not sure how you are building your dynamic schemas at the moment, but there is `ITypeModule` which also supports invalidation: https://chillicream.com/blog/2021/09/27/hot-chocolate-12#dynamic-schemas

You can find a full example here: https://github.com/ChilliCream/hotchocolate-examples/tree/master/misc/TypeModules

**Reply by jimitndiaye** [reply]
Thanks. Those links look like exactly what I need.
